### PR TITLE
Fix RNAseq data loading SQL syntax error.

### PR DESCRIPTION
**Pull request by ShifaSZ**
A fix for bug #3264.

### CODE REVIEW

File: seqr/views/utils/dataset_utils.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
     sample_projects = Project.objects.filter(family__individual__sample__in=samples_to_load.keys()).values(
         'guid', 'name', new_sample_ids=ArrayAgg(
             'family__individual__sample__sample_id', distinct=True, ordering='family__individual__sample__sample_id',
-            filter=~Q(family__individual__id__in=prev_loaded_individual_ids)
+            filter=~Q(family__individual__id__in=prev_loaded_individual_ids) if prev_loaded_individual_ids else None
```
Comment (ShifaSZ) [inline seqr/views/utils/dataset_utils.py:438]: When `prev_loaded_individual_ids` is empty, the SQL query generated by Django is incorrect (with an empty condition in the `where` clause).

**Comment by hanars** [comment]
this needs a unit test that would have caught the error, so we don;t accidentally reintroduce this issue in the future

File: seqr/views/apis/data_manager_api_tests.py [hunk 2]
```diff
@@ -816,6 +816,14 @@ def mock_write(content):
                 mock_open.assert_called_with(file_name, 'wt')
                 self.assertListEqual(mock_writes, [row.replace(PLACEHOLDER_GUID, new_sample_guid) for row in params['parsed_file_data']])
 
+                # test loading new data without existing data
+                data = [params['new_data'][0]]
+                data[0][0] = 'NA19678'  # load data for a new individual
+                _set_file_iter_stdout([header] + data)
+                body.pop('mappingFile')
+                response = self.client.post(url, content_type='application/json', data=json.dumps(body))
+                self.assertEqual(response.status_code, 200)
```
Comment (hanars) [inline seqr/views/apis/data_manager_api_tests.py:825]: You need to add tests that this is actually doing what we want, not just that it has a successful status code. Add a test for the `response_json` being what we expect and use `_check_rna_sample_model` to test the database mode is correct

**Review by hanars — CHANGES_REQUESTED** [review]

File: seqr/views/apis/data_manager_api_tests.py (hunk 2 above)
Comment (ShifaSZ) [reply seqr/views/apis/data_manager_api_tests.py:825]: When the bug happens, there is an uncaught exception. So the `response.status_code` will be 500. For testing the bug itself, it is enough to test the status code. But I will add the model check to ensure the change is correct.

Comment (hanars) [reply seqr/views/apis/data_manager_api_tests.py:825]: tests are to test functionality for all changes going forward, not just one bug this one time. What if someone changed fixture data and this no longer was doing what we wanted it to do and was instead returning a 200 response with a warning saying everything in the file was skipped? How would we know from this test that we had lost coverage?

File: seqr/views/apis/data_manager_api_tests.py [hunk 3]
```diff
@@ -816,6 +816,34 @@ def mock_write(content):
...
+                # test loading new data without existing data
+                mock_logger.reset_mock()
+                data = [params['new_data'][3]]
+                data[0][0] = 'NA19678'  # load data for a new individual
+                _set_file_iter_stdout([header] + data)
+                body.pop('mappingFile')
+                response = self.client.post(url, content_type='application/json', data=json.dumps(body))
+                self.assertEqual(response.status_code, 200)
+                info = [
+                    f'Parsed 1 RNA-seq samples',
```
Comment (hanars) [inline seqr/views/apis/data_manager_api_tests.py:828]: this does not need to be an f string, its failing codacy

**Review by hanars — CHANGES_REQUESTED** [review]

**Review by hanars — APPROVED** [review]

**Pull request merged**